tests/Task.test.ts

```diff
@@ -603,4 +631,41 @@ describe('toggle done', () => {
             todaySpy.mockClear();
         },
     );
+
+    it('supports recurrence rule after a due date', () => {
+        // Arrange
+        const line =
+            '- [ ] this is a task 🗓 2021-09-12 ✅ 2021-06-20 🔁 every day';
```

Comment:
Is there a significance to the Done date being present, in an incomplete task?

Reply:
> Is there a significance to the Done date being present, in an incomplete task?

This is why I experimented with the current release to see what happens when completing a recurring task that already had a done date - which I wrote up in https://github.com/obsidian-tasks-group/obsidian-tasks/discussions/851.

But unless it's intentional that the done date is in the input of this test, it might be better to just delete it, to avoid any confusion???

Reply:
There's no significance, I just copied this line from a different test but made the task incomplete, hence this result :)
Removed the done date, as it is not relevant to this test.